.github/workflows/scripts/jira_helper.py

```diff
@@ -78,24 +87,35 @@ def _get_gh_issue_comments(self, url):
 
     def _issue_helper(self):
         event_name = os.environ['EVENT_NAME']
+        issue_state = get_issue_state(os.environ['ISSUE_STATE'])
         # Make sure the issue is closed
-        issue_is_closed = event_name == "closed" or event_name == "deleted"
+        closed_event = event_name == "closed" or event_name == "deleted"
         # If re-opened, then the issue should transition to backlog
         issue_reopened = event_name == "reopened"
-        issue_edited = event_name == "edited" or event_name == "labeled" or event_name == "unlabeled"
+        issue_edited = event_name == "edited"
+        issue_labeled = event_name == "labeled" or event_name == "unlabeled"
```

Comment:
As this script is more complicated I highly encourage you to do something like the following. The variability of all the conditionals here make it very hard to follow the logic personally.

```python
if event_name == "opened":
  create_issue()
elif event_name == "edited":
  issue = find_issue()
  issue = create_issue() if not issue else issue
  update_issue(issue)
elif event_name == "labeled" or event_name == "unlabeled":
  issue = find_issue()
  update_issue(issue) if issue
elif event_name == "closed":
  issue = find_issue()
  close_issue(issue) if issue
else:
  raise Exception("Unknown event!")
```

Reply:
done thanks for the suggestion